cloudferrylib/os/compute/instances.py

```diff
@@ -29,6 +29,20 @@ def update_user_ids_for_instance(db, instance_id, user_id):
     db.execute(sql)
 
 
+def get_flav_details(db, instance_id):
+    sql = ("SELECT vcpus,memory_mb,root_gb,ephemeral_gb "
+           "FROM nova.instances "
+           "WHERE nova.instances.uuid = '{instance_id}' "
+           "AND NOT nova.instances.vm_state = 'deleted';").format(
```

Comment:
Not 100% sure, but it might be better to use `nova.instances.deleted = 0` check instead of `nova.instances.vm_state = 'deleted'`